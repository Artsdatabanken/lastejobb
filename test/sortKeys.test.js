const { json } = require("lastejobb");

const input = {
  z: 2,
  k: [6, 1, 3],
  y: { c: 5, a: "abc" }
};
const inputArray = [6, 1, 3];

// Recursively sort objects
test("sort JSON object", () => {
  const actual = json.sortKeys(input);
  expect(actual).toMatchSnapshot();
});

// We don't sort the array
test("sort JSON array", () => {
  const actual = json.sortKeys(inputArray);
  expect(actual).toMatchSnapshot();
});
