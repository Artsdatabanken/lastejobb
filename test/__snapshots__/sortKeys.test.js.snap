// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sort JSON array 1`] = `
Array [
  6,
  1,
  3,
]
`;

exports[`sort JSON object 1`] = `
Object {
  "k": Array [
    6,
    1,
    3,
  ],
  "y": Object {
    "a": "abc",
    "c": 5,
  },
  "z": 2,
}
`;
